File: langs/python3/problems/workers/stub.py
class Simulation:
    def __init__(self):
        pass

    def add_worker(self, worker_id, position, compensation):
        """Returns "true" if created, "false" if the id exists."""
        raise NotImplementedError

    def register(self, worker_id, timestamp):
        """Toggle in or out. "registered", or "invalid_request" if missing."""
        raise NotImplementedError

    def get(self, worker_id):
        """Total finished time as a string, or "" if missing."""
        raise NotImplementedError

    def top_n_workers(self, n, position):
        """Current position as id(time), ... Finished time in that position.

        Example: "Jason(50), John(50), Ashley(0)".
        """
        raise NotImplementedError

    def promote(self, worker_id, new_position, new_compensation, start_timestamp):
        """Queue one change. Applied on the next enter at or after start.

        Returns "success" or "invalid_request".
        """
        raise NotImplementedError

    def calc_salary(self, worker_id, start_timestamp, end_timestamp):
        """Pay for finished sessions overlapping the window, or ""."""
        raise NotImplementedError

File: langs/java/problems/workers/stub.java
public class Simulation {
    public Simulation() {}

    /**
     * Returns {@code "true"} if created, {@code "false"} if the id exists.
     */
    public String addWorker(String workerId, String position, int compensation) {
        return "";
    }

    /**
     * Toggle in or out. Returns {@code "registered"}, or
     * {@code "invalid_request"} if missing.
     */
    public String register(String workerId, int timestamp) {
        return "";
    }

    /**
     * Total finished time as a string, or {@code ""} if missing.
     */
    public String get(String workerId) {
        return "";
    }

    /**
     * Current position as {@code id(time), ...}. Time is finished time
     * in that position. Example: {@code "Jason(50), John(50), Ashley(0)"}.
     */
    public String topNWorkers(int n, String position) {
        return "";
    }

    /**
     * Queue one change. Applied on the next enter at or after
     * startTimestamp. Returns {@code "success"} or
     * {@code "invalid_request"}.
     */
    public String promote(
            String workerId, String newPosition, int newCompensation, int startTimestamp) {
        return "";
    }

    /**
     * Pay for finished sessions overlapping the window, or {@code ""}.
     */
    public String calcSalary(String workerId, int startTimestamp, int endTimestamp) {
        return "";
    }
}
